File: client/src/main/java/au/com/codeka/warworlds/client/store/BaseStore.java
package au.com.codeka.warworlds.client.store;

import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/** Base class for "store" class. */
public abstract class BaseStore {
  private final String name;
  private final SQLiteOpenHelper helper;

  public BaseStore(String name, SQLiteOpenHelper helper) {
    this.name = name;
    this.helper = helper;
  }

  /**
   * Called by the {@link DataStore} when the database is created.
   */
  public abstract void onCreate(SQLiteDatabase db);

  /**
   * Called by the {@link DataStore} when we need to upgrade. By default, does nothing.
   */
  public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/store/BaseStore.kt
package au.com.codeka.warworlds.client.store

import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/** Base class for "store" class.  */
abstract class BaseStore(private val name: String, private val helper: SQLiteOpenHelper) {
  /**
   * Called by the [DataStore] when the database is created.
   */
  abstract fun onCreate(db: SQLiteDatabase)

  /**
   * Called by the [DataStore] when we need to upgrade. By default, does nothing.
   */
  open fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}

}